packages/viewer/src/components/viewer-pin-tool/utils.ts

```diff
@@ -29,13 +29,39 @@ export function translatePointToScreen(
  */
 export function translatePointToRelative(
   pt: Point.Point,
-  canvasDimensions: Dimensions.Dimensions
+  canvasDimensions: Dimensions.Dimensions,
+  includeOffset?: boolean
 ): Point.Point {
+  const verticalScaleFactor = 1 / canvasDimensions.height;
+  const horizontalScaleFactor = 1 / canvasDimensions.width;
+  const canvasCenterPoint = Dimensions.center(canvasDimensions);
+
   const point = Point.scale(
-    Point.subtract(pt, Dimensions.center(canvasDimensions)),
-    1 / canvasDimensions.width,
-    1 / canvasDimensions.height
+    Point.subtract(pt, canvasCenterPoint),
+    horizontalScaleFactor,
+    verticalScaleFactor
   );
 
-  return constrainRelativePoint(point, Range.create(-0.5, 0.5));
+  if (includeOffset) {
+    const offset = Point.scale(
+      Point.create(20, 20),
+      horizontalScaleFactor,
+      verticalScaleFactor
+    );
+
+    // We want to place the label towards the center of the screen.
+    // The given point corresponds to the upper left corner of the label, so increase the offset
+    // when placing it to the left or above to ensure that part of the label line is visible
+    const centerIsToTheRightInXDirection = canvasCenterPoint.x > pt.x;
+    const xOffset = centerIsToTheRightInXDirection ? offset.x : -3 * offset.x;
+
+    const centerIsBelowInYDirection = canvasCenterPoint.y > pt.y;
+    const yOffset = centerIsBelowInYDirection ? offset.y : -3 * offset.y;
+
+    const offsetPoint = Point.add(point, Point.create(xOffset, yOffset));
+
+    return constrainRelativePoint(offsetPoint, Range.create(-0.5, 0.5));
```

Comment:
Wondering if we could make these offset values configurable from the component? Seems like it would be nice to have a bit of control over how far the pin label is from the initial point. Maybe something that could be read in from a CSS variable? I.e. `viewer-pin-tool-initial-label-offset-x: 20px` or something similar.

Additionally, any specific reason this is scaled by `-3` if it needs to flip to the other side? Thinking we could run into issues with a constant there if the label starts off with a larger width.

Reply:
I added a CSS variable, good idea. 

The point that this returns corresponds to the upper left corner of the label, so if the label is placed above or to the left, the label line is smaller since the distance from the anchor point to the upper left corner of the label crosses the body of the label. Multiplying the offset value by 3 helps to ensure that we have a visible label line when a pin is first placed.  Since it's relative to the value of the CSS variable, I think a user could adjust that variable if they also adjusted the initial width of the label.